llvm/test/CodeGen/AArch64/neon-bitwise-instructions.ll

```diff
@@ -1100,9 +1100,7 @@ define <8 x i8> @vselect_constant_cond_zero_v8i8(<8 x i8> %a) {
 ; CHECK-GI:       // %bb.0:
 ; CHECK-GI-NEXT:    adrp x8, .LCPI83_0
 ; CHECK-GI-NEXT:    ldr d1, [x8, :lo12:.LCPI83_0]
-; CHECK-GI-NEXT:    shl v1.8b, v1.8b, #7
-; CHECK-GI-NEXT:    sshr v1.8b, v1.8b, #7
-; CHECK-GI-NEXT:    and v0.8b, v0.8b, v1.8b
+; CHECK-GI-NEXT:    and v0.8b, v1.8b, v0.8b
```

Comment:
Here are improvements.

Reply:
It is an interplay between this combiner and the imported code. This combiner is too weak. Trying to make it more powerful will always be a compromise. Once it get's more powerful there will be less compromises. Never going to touch this combiner again will be a sad state.